napari/layers/labels/labels.py

```diff
@@ -525,7 +542,54 @@ def paint(self, coord, new_label, refresh=True):
             slice_coord = tuple(slice_coord)
 
         # update the labels image
-        self.data[slice_coord] = new_label
+
+        if self._overwrite:
```

Comment:
I wonder if we should also use this option when the brush is set to be the "background_label" as otherwise when painting with the background label nothing happens in non-overwrite mode